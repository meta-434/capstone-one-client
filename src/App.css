@import url('https://rsms.me/inter/inter.css');
  html { font-family: 'Inter', sans-serif; }
@supports (font-variation-settings: normal) {
  html { font-family: 'Inter var', sans-serif; }
}
@supports (font-variation-settings: normal) {
  html { font-family: 'Inter Bold', sans-serif; }
}

* {
  box-sizing: border-box;
  font-family: 'Inter';
  background: url('images/background.png') no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

body {
  font-size: 20px;
  color: #444;
  text-align: center;
  line-height: 1.5;
}

nav,
footer {
  padding: 1% 0;
}

.error {
  margin-top: 140px;
}

nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
  height: 70px;
}

header {
  /*padding: 45px 15px;*/
  margin-top: 70px;
}

section {
  min-height: 200px;
  padding: 30px 15px;
  color: #5E81AC;
}

p {
  margin: 0 auto;
  max-width: 650px;
}

p + p {
  margin-top: 15px;
}

button:disabled {
  border: 2px #4C566A solid;
  border-radius: 3px;
  color: #4C566A;
}

button {
  border: 2px inherit solid;
  border-radius: 3px;
}