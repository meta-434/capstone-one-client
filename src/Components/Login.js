import React, { Component } from 'react';
import './Login.css'

class Login extends Component {
    render() {
        return(
            <header>
                Login placeholder.
            </header>
        );
    }
}

export default Login;