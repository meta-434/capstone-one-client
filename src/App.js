import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Nav from './Components/Nav/Nav.js';
import Login from './Components/Login/Login.js';
import Signup from './Components/Signup/Signup.js';
import Main from './Components/Main/Main.js';
import Notes from './Components/Notes/Notes.js';
import Sessions from './Components/Sessions/Sessions.js';
import TimerComponent from './Components/TimerComponent/TimerComponent.js';
import PomodoroContext from './PomodoroContext';
import './App.css'

class App extends Component {
  state = {
    sessions: [],
    notes: [],
    authToken: undefined,
    error: undefined,
  }

  logOut = () => {
    this.setState({authToken: undefined});
    sessionStorage.clear();
  }

  handlePostSignup = ({ username, password }) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/signup`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        'username': username,
        'password': password,
      })
    })
        .then(result => {
            let json = result.json() // There's always a response body
            if (result.status >= 200 && result.status < 300) { return json }
            return json.then(Promise.reject.bind(Promise))

        })
        .then(result =>  result.json())
        .catch(error => {
          this.setState({error});
        });
  };

    handlePostAuthenticate = ({ username, password }) => {
        console.log('for login');
        fetch(process.env.REACT_APP_SERVER_URL + `/authenticate`, {
            method: 'post',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                'username': username,
                'password': password,
            })
        })
            .then(result => {
                let json = result.json() // There's always a response body
                if (!(result.status >= 200 && result.status < 300)) {
                    return json.then(Promise.reject.bind(Promise))
                }
                return json;
            })
            .then(resJson => {
                if (!!resJson.token) {
                    this.setState({ authToken: resJson.token, username })
                    sessionStorage.setItem('access-token', this.state.authToken);
                    sessionStorage.setItem('username', username);
                    this.handleGetSessions();
                    this.handleGetNotes();
                }
                else {
                    throw new Error(' error in authenticating. check username and password. ');
                }
            })
            .catch(error => {
                console.error(error);
                this.setState({error});
            });
    };

  handlePostSession = ({session_name, session_description, ownerId}) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/sessions`, {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      },
      body: JSON.stringify({
        session_name,
        session_description,
        session_owner: ownerId
      })
    })
        .then(res => this.handleGetSessions())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleGetSessions = () => {
    fetch(process.env.REACT_APP_SERVER_URL+ `/api/sessions`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(response => response.json())
        .then(responseJson => {
              if (responseJson.success && responseJson.success === false) {
                throw new Error('error in getting sessions')
              } else {
                this.setState({
                  sessions: responseJson
                })
              }
            }
        )
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleDeleteSession = (id) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/sessions/${id}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(res => this.handleGetSessions())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handlePostNote = ({note_name, note_content, ownerId}) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/notes`, {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      },
      body: JSON.stringify({
        note_name,
        note_content,
        note_owner: ownerId
      })
    })
        .then(res => this.handleGetNotes())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleGetNotes = () => {
    fetch(process.env.REACT_APP_SERVER_URL+ `/api/notes`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(response => response.json())
        .then(responseJson => {
              if (responseJson.success && responseJson.success === false) {
                throw new Error('error in getting notes')
              } else {
                this.setState({
                  notes: responseJson
                })
              }
            }
        )
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleDeleteNote = (id) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/notes/${id}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(res => this.handleGetNotes())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  render () {
    const { sessions, notes, authToken, username, error } = this.state;
    const context = {
      sessions,
      notes,
      username,
      authToken,
      error,
      handlePostAuthenticate: this.handlePostAuthenticate,
      handlePostSignup: this.handlePostSignup,
      logOut: this.logOut,
      handleGetNotes: this.handleGetNotes,
      handlePostNote: this.handlePostNote,
      handleDeleteNote: this.handleDeleteNote,
      handleGetSessions: this.handleGetSessions,
      handleDeleteSession: this.handleDeleteSession,
      handlePostSession: this.handlePostSession,
    }
    return (
        <PomodoroContext.Provider value={(context)}>
          <div className="app">
            <Route path="/" render={routeProps => <Nav {...routeProps}/>}/>
            <main className="app-content">
              <Route exact path="/" component={Main} />
              <Route path="/login" component={Login}/>
              <Route path="/signup" component={Signup} />
              <Route path="/timer" component={TimerComponent}/>
              <Route path="/sessions/:sessionId?" component={Sessions}/>
              <Route path="/notes/:noteId?" component={Notes}/>

            </main>
          </div>
        </PomodoroContext.Provider>
    );
  }
}

export default App;