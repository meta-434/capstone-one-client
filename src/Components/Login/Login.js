import React, { Component } from 'react';
import './Login.css'
import PomodoroContext from "../../PomodoroContext";
import ErrorDisplay from "../ErrorDisplay/ErrorDisplay";

class Login extends Component {

    static contextType = PomodoroContext;

    constructor() {
        super();
        this.state = {
            username: undefined,
            password: undefined,
            validation: '',
            usernameValid: false,
            passwordValid: false,
            error: undefined,
        }
    }

    componentDidMount() {
        this.validateUsername(this.state.username);
        this.validatePassword(this.state.password);
    }

    componentDidUpdate(prevProps, prevState, snapshot) {
        if (!!this.context.authToken || sessionStorage[`access-token`]) {
            this.context.clearError();
            this.props.history.push('/timer');
        }
    }

    handlePostSubmit = (e) => {
        e.preventDefault();
        this.context.handlePostAuthenticate(this.state);
    }

    handleUsername = (e) => {
        let usernameInput = e.target.value;
        this.setState({ username: usernameInput }, () => this.validateUsername(usernameInput));
    }

    handlePassword = (e) => {
        let passwordInput = e.target.value;
        this.setState({ password: passwordInput }, () => this.validatePassword(passwordInput));
    }

    validateUsername = (username) => {
        let validationMessages;
        let hasError = false;

        if (!username) {
            hasError = true;
            validationMessages = ' username cannot be blank '
        }
        else {
            validationMessages = '';
        }

        this.setState({
            usernameValid: !hasError,
            usernameValidation: validationMessages,
        }, () => this.usernameValid(username))
    }

    validatePassword = (password) => {
        let validationMessages;
        let hasError = false;

        if (!password) {
            hasError = true;
            validationMessages = ' password cannot be blank '
        }
        else {
            validationMessages = '';
        }

        this.setState({
            passwordValid: !hasError,
            passwordValidation: validationMessages,
        }, () => this.passwordValid(password))
    }

    usernameValid = (username) => {
        if (this.state.username) {
            this.setState({username});
        }
    }

    passwordValid = (password) => {
        if (this.state.password) {
            this.setState({password})
        }
    }

    render() {
        return(
            <>
                {
                    (!!this.context.error)
                    ? <ErrorDisplay />
                    : ''
                }
                <main className="login-form">
                    <section>
                        <header>
                            login to access saved sessions and notes!
                        </header>
                    </section>
                    <section>
                        <form
                            className='react-form'
                            onSubmit={this.handlePostSubmit}>
                            <label htmlFor='login-username'>username: </label>
                            <input
                                type="text"
                                id="login-username"
                                name="login-username"
                                className="login-username"
                                onChange={this.handleUsername}
                                placeholder={'enter username'}
                                aria-label="username"
                                aria-required="true"
                                aria-describedby="error-box"
                            />
                            <label htmlFor='login-password'>password: </label>
                            <input
                                type="text"
                                id="login-password"
                                name="login-password"
                                className="login-password"
                                onChange={this.handlePassword}
                                placeholder={'enter password'}
                                aria-label="password"
                                aria-required="true"
                                aria-describedby="error-box"
                            />
                            <button
                                className="submit-button"
                                type="submit"
                                disabled={!this.state.usernameValid || !this.state.passwordValid }>
                                Submit
                            </button>
                            <section className="error-box" id="error-box" aria-live="assertive">
                                {this.state.usernameValidation}
                                <br />
                                {this.state.passwordValidation}
                            </section>
                        </form>
                    </section>
                </main>
            </>
        );
    }
}

export default Login;