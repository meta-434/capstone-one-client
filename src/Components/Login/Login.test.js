import React from 'react';
import { render } from '@testing-library/react';
import Login from './Login';
import App from '../../App';
import {BrowserRouter} from "react-router-dom";
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PomodoroContext from "../../PomodoroContext";

test('renders Login component without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><App><Login /></App></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the Login UI as expected', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <App>
                    <PomodoroContext>
                        <Login />
                    </PomodoroContext>
                </App>
            </BrowserRouter>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
})