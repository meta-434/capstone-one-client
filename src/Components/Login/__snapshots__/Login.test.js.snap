// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Login UI as expected 1`] = `
<div
  className="app"
>
  <nav>
    <h1
      className="nav-h1"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Pomodoro Timer
      </a>
    </h1>
    <div
      className="button-container"
    >
      <button>
        <a
          href="/timer"
          onClick={[Function]}
        >
          Timer
        </a>
      </button>
      <button>
        <a
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
      </button>
      <button>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </button>
    </div>
  </nav>
  <main
    className="app-content"
  >
    <section>
      <p>
        
      </p>
      <header>
        <h3>
          What is the Pomodoro Technique ?
        </h3>
      </header>
      <p>
        <b>
          <em>
            To try out the app, log in with the following credentials
          </em>
        </b>
        <br />
        username:
        <b>
          <em>
             test-user 
          </em>
        </b>
        <br />
        password:
        <b>
          <em>
             test-password 
          </em>
        </b>
      </p>
      <p>
        The Pomodoro Technique is a time-management technique that encourages productivity and focus
      </p>
      <br />
      <img
        alt="timer"
        src="timer.png"
      />
      <p
        className="caption"
      >
        Timer Page
      </p>
      <p>
        By working on a specific task in 25 minute increments, with 5 minute breaks, your work sessions never lack in direction. Every 4 work sessions, take a longer break. 20 minutes, maybe 30. This time will help you rest before you next round.
         
      </p>
    </section>
    <section>
      <header>
        <h3>
          Track your Pomodoro sessions
        </h3>
      </header>
      <img
        alt="sessions page"
        src="sessions.png"
      />
      <p
        className="caption"
      >
        Session Page
      </p>
      <p>
        By reviewing tracking your Pomodoro sessions, you increase your accountability - you want to be able to put something down as completed. By referring to your past Pomodoro sessions, you can see your progress as you work through your sessions.
         
      </p>
    </section>
    <section>
      <header>
        <h3>
          Keep Useful Notes
        </h3>
      </header>
      <img
        alt="notes page"
        src="notes.png"
      />
      <p
        className="caption"
      >
        Note Page
      </p>
      <p>
        Keep formulae, phone numbers, or just encouraging thoughts - your most recent note will be displayed on the timer page for reference during work sessions. Notes can only be modified from the notes tab as to not distract the user during work sessions. Remember to set your notes before you start the timer!
      </p>
    </section>
  </main>
</div>
`;
